contracts/DecentralizedKVMinable.sol

```diff
@@ -133,12 +133,18 @@ contract DecentralizedKVMinable is DecentralizedKV {
         }
     }
 
+    /* TODO: THIS IS VULNERABLE WHEN FACING LAZY DATA PROVIDER
+       It can be affected by an attack vertor, for the way generates random access positions and verifying procedure
```

Comment:
Is it because attackers can precompute all the indexes to check out whether they actually have them before doing any real io reading?

Reply:
Yes. As a comment, for an honest miner, he/she also needs to use data and nonce to compute, for checking whether they can pass PoRA check, or Hashimoto-Dagger work proof. To me, the key point of PoRA vulnerability comes from the fact that, the value to use hash generating next entry to examine, is not that random at all. Because miner always hold data, he/she can think up a series of data slice, passing the PoRA without much effort.